Add Redis helper library (#125)
Automattic/vip-go-node: src/redis/README.md

# VIP Go Redis Helper Library

## Initialization

Instantiate a new Redis client:

``` js

> const { redis } = require( '@automattic/vip-go' );

Reviewer: I think we still need to expose `redis` in https://github.com/Automattic/vip-go-node/blob/master/src/index.js in order to be able to import it like this.

In addition, I see we are exporting a `new Redis()` here https://github.com/Automattic/vip-go-node/pull/125/files#diff-66b52dd10fa44ca40f44c06793a2aec1R77 but I guess in order to give the client the possibility to pass their own logger, we should just export the `Redis` class, and they need to do the `new Redis ( { logger: myLogger } );` on their own.

Or if we can avoid exporting classes, that would be good as all our other helpers (logger, newrelic, server) do not use a class approach when used by clients. It would be good to keep a consistency of:

```
const { logger, server, newrelic, redis } = require( '@automattic/vip-go-node' );

logger();
server();
newrelic();
redis();
```

We can use the class approach in the helper itself, but just exposing a function instead to keep the consistency. Hope I've been clear, if it's not the case, please let me know

const redis = new Redis();
```

> By default, this helper library logs to the console. Pass in your own logger like so:

Reviewer: ```suggestion
By default, this helper library logs to the `console`. If you prefer, you can set your own logger:
```

Author: Changed 👍

```js

> redis( { logger: yourLogger } );

Reviewer: ```suggestion
const redis = new Redis( { logger: yourLogger } );
```

Author: 🤦‍♀duh. Thank you!

```

## Configuration
Configuring a Redis client compatible with the VIP Go platform requires a `REDIS_MASTER` environment variable. This is the host and port of your Redis server, which might look something like this:
`REDIS_MASTER = 123.456.789.123:3000`

If you require a password, please set the `REDIS_PASSWORD` environment variable as well.

## Reconnecting to Redis
If the connection to the Redis server is lost, offline queuing for commands will be disabled.

> After x amount of reconnection attempts, all pending commands will be flushed so that there are no stale commands.

Reviewer: ```suggestion
After 3 reconnection attempts, all pending commands will be flushed so that there are no stale commands.
```

Author: Updated 👍

> You may set the number of attempts with the `QUEUED_CONNECTION_ATTEMPTS` environment variable, or it will default to three reconnection attempts.

Reviewer: ```suggestion
You can change this number by using the `QUEUED_CONNECTION_ATTEMPTS` environment variable.
```

Author: Changed 👍
